include/spdlog/sinks/qplaintextedit_sink.h

```diff
@@ -0,0 +1,71 @@
+// Copyright(c) 2015-present, Gabi Melman, mguludag and spdlog contributors.
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+
+#pragma once
+
+//
+// Custom sink for QPlainTextEdit and its childs(QTextBrowser... etc)
+// Building and using requires Qt library.
+//
+
+#include "spdlog/common.h"
+#include "spdlog/details/log_msg.h"
+#include "spdlog/details/synchronous_factory.h"
+#include "spdlog/sinks/base_sink.h"
+
+#include "qplaintextedit_sink_p.h"
+
+namespace spdlog {
+namespace sinks {
+template <typename Mutex>
+class qplaintextedit_sink : public base_sink<Mutex> {
+public:
+    qplaintextedit_sink(QPlainTextEdit *textedit = nullptr) {
+        if (textedit != nullptr) {
+            textedit_p = std::make_shared<_spdlog_p::_sinks_p::qplaintextedit_sink_p>(textedit);
+        } else {
+            throw spdlog_ex("Error opening QPlainTextEdit");
+        }
+    }
+
+    ~qplaintextedit_sink() { flush_(); }
+
+protected:
+    void sink_it_(const details::log_msg &msg) override {
+        memory_buf_t formatted;
+        base_sink<Mutex>::formatter_->format(msg, formatted);
+        auto str = std::string(formatted.begin(), formatted.end() - 2);

```

Comment:
Is there a way to avoid the expensive creation of std string? maybe use a string view?

Reply:
Ok. I changed it to `spdlog::string_view_t` but widget's slots accepts `QString` so constructed `QString` and emit its signal.